### PR TITLE
Fix format to show contents properly

### PR DESCRIPTION
**Pull request by Muhammad-Yazdian**
Lots of contents were not displayed correctly. I put some empty lines and :math: labels to fixed it.

Documentation of move to pose algorithm #596

### CODE REVIEW

**Comment by AtsushiSakai** [comment]
Let's close and open to run CI again.

**Comment by AtsushiSakai** [comment]
Hmm weird. Circle CI does not run, but it works in #609. But it's ok, we can check the doc in local.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 1]
```diff
@@ -13,34 +13,39 @@ Constructor
...
 Constructs an instantiate of the PathFinderController for navigating a 3-DOF wheeled robot on a 2D plane.
 
 Parameters:
-- Kp_rho: The linear velocity gain to translate the robot along a line towards the goal
-- Kp_alpha: The angular velocity gain to rotate the robot towards the goal
-- Kp_beta: The offset angular velocity gain accounting for smooth merging to the goal angle (i.e., it helps the robot heading to be parallel to the target angle.)
+
+- | :math:`Kp_rho` : The linear velocity gain to translate the robot along a line towards the goal
+- | :math:`Kp_alpha` : The angular velocity gain to rotate the robot towards the goal
+- | :math:`Kp_beta` : The offset angular velocity gain accounting for smooth merging to the goal angle (i.e., it helps the robot heading to be parallel to the target angle.)
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:25]: The math does not rendered properly, maybe you should `Kp_{rho}`? Please check other math renderings.
SS 2021-12-26 at 10 42 46

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 2]
```diff
@@ -13,34 +13,39 @@ Constructor
 ~~~~~~~~~~~
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:13]: I merged your PR about adding gif file. So, you can add the link of it.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 3]
```diff
@@ -13,34 +13,39 @@ Constructor
...
 
 .. code-block:: ipython3
+
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:16]: I think the most variable doc for users is about the control logic and methematics of the PathFinderController which are implemented here:
https://github.com/AtsushiSakai/PythonRobotics/blob/7034d5ff9eb19eff561f84c806ec591b2d0c8e26/Control/move_to_pose/move_to_pose.py#L73-L78
Is it possible to add the doc here?

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 1 above)
Comment (Muhammad-Yazdian) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst]: Interesting. What I have in the preview is different form yours. To test it, I just included {rho} in line 23. The picture below the subscripts with _{...} and without _{...}. If _{...} fixes it on the final online doc, I will apply it to all other subscripts too.

image

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 2 above)
Comment (Muhammad-Yazdian) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:13]: I added the gif link and removed the png.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 1 above)
Comment (AtsushiSakai) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst]: Current code show this in my enviroment:
SS 2021-12-27 at 19 45 38
 Maybe, your environment has an issue about `sphinx.ext.mathjax' .

**Comment by AtsushiSakai** [comment]
Let's close and open to run CI again.

**Comment by AtsushiSakai** [comment]
Doc build is back in CI. Please check `build_doc artifact` Details link.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 1 above)
Comment (Muhammad-Yazdian) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst]: I decided to make it simple and used bold text for the parameter definitions.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 3 above)
Comment (Muhammad-Yazdian) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:28]: The document for the logic added.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 4]
```diff
@@ -1,57 +1,107 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The distance between the robot and the goal position, :math:`\rho`, is computed as
+
+.. math::
+ \rho = \sqrt{(x_{robot} - x_{target})^2 + (y_{robot} - y_{target})^2}.
+
+The distance :math:`\rho` is used to determine the robot speed. The idea is to slow down the robot as it gets closer to the target.
+
+.. math::
+ \color{Red} {v = K_P{_\rho} \times \rho} (Eq. 1)
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:72]: Why this equation is red?
SS 2022-01-18 at 21 07 24

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 5]
```diff
@@ -1,57 +1,107 @@
...
+
+Note that for your applications, you need to tune the speed gain, :math:`K_P{_\rho}` to a proper value.
+
+To turn the robot and align its heading, :math:`\theta`, toward the target position (not orientation),  :math:`\rho \vec{u}`, we need to compute the angle difference :math:`\alpha`. 
+
+.. math::
+ \alpha = (\arctan2(y_diff, x_diff) - \theta + \pi) mod (2\pi) - \pi
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:79]:
```suggestion
 \alpha = (\arctan2(y_{diff}, x_{diff}) - \theta + \pi) mod (2\pi) - \pi
```

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 6]
```diff
@@ -1,57 +1,107 @@
...
+ \beta = (\theta_{goal} - \theta - \alpha + \pi) mod (2\pi) - \pi
+
+Note that to cancel out the effect of :math:`\alpha` when the robot is at the vicinity of the target, the term 
+
+:math:`-\alpha` is included.
+
+The final angular velocity command is given by
+
+.. math::
+ \color{Red} {w = K_P{_\alpha} \alpha - K_P{_\beta} \beta} (Eq. 2)
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:95]: Why this equation is red?

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 7]
```diff
@@ -1,57 +1,107 @@
...
+
+This section explains the logic of a position controller for systems with constraint (non-Holonomic system).
+
+The position control of a 1-DOF (Degree of Freedom) system is quite straightforward. We only need to compute a position error and multiply it with a proportional gain to create a command. The actuator of the system takes this command and drive the system to the target position. This controller can be easily extended to higher dimensions (e.g., using Kp_x and Kp_y gains for a 2D position control). In these systems, the number of control commands is equal to the number of degrees of freedom (Holonomic system). 
+
+To describe the configuration of a car on a 2D plane, we need three DOFs (i.e., x, y, and theta). But to drive a car we only need two control commands (theta_engine and theta_steering_wheel). This difference is because of a constraint between the x and y DOFs. The relationship between the delta_x and delta_y is governed by the theta_steering_wheel.
+
+Note that a car is normally a non-Holonomic system but if the road is slippery, the car turns into a Holonomic system and thus it needs three independt commands to be controlled.
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:10]:
```suggestion
Note that a car is normally a non-Holonomic system but if the road is slippery, the car turns into a Holonomic system and thus it needs three independent commands to be controlled.
```

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]
Thank you for fixing doc. I have some comments and questions.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 4 above)
Comment (Muhammad-Yazdian) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:72]: I made v an omega red to emphasize that they are the output of the algorithm. I tried to draw a box around them but unfortunately no luck. Should we just stick with simple black color?

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 6 above)
Comment (Muhammad-Yazdian) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst]: I changed both back to black.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 8]
```diff
@@ -1,57 +1,109 @@
...
+
+Note that to cancel out the effect of :math:`\alpha` when the robot is at the vicinity of the target, the term 
+
+:math:`-\alpha` is included.
+
+The final angular speed command is given by
+
+.. math::
+ \omega = K_P{_\alpha} \alpha - K_P{_\beta} \beta\qquad (Eq. 2)
+
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:96]: Maybe this space breaks the rendering:
SS 2022-01-20 at 21 19 01

```suggestion

```

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst [hunk 9]
```diff
@@ -1,57 +1,109 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The distance between the robot and the goal position, :math:`\rho`, is computed as
+
+.. math::
+ \rho = \sqrt{(x_{robot} - x_{target})^2 + (y_{robot} - y_{target})^2}.
+
+The distance :math:`\rho` is used to determine the robot speed. The idea is to slow down the robot as it gets closer to the target.
+
+.. math::
+ v = K_P{_\rho} \times \rho\qquad (Eq. 1)
```
Comment (AtsushiSakai) [inline docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:72]: Cloud you use `:label:` for math labeling? This doc might help you: [doc](https://github.com/AtsushiSakai/PythonRobotics/blob/master/docs/modules/path_planning/quintic_polynomials_planner/quintic_polynomials_planner.rst)
SS 2022-01-20 at 21 23 32

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]
This is the last review, I will merge this PR after you address these.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 8 above)
Comment (Muhammad-Yazdian) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst:96]: The leading space on line 97 was the issue. I fixed it.

File: docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst (hunk 9 above)
Comment (Muhammad-Yazdian) [reply docs/modules/control/move_to_a_pose_control/move_to_a_pose_control.rst]: Thanks, that helped a lot!

**Review by AtsushiSakai — APPROVED** [review]
LGTM. Thank you for great doc improvement!!

**Comment by Muhammad-Yazdian** [comment]
@AtsushiSakai It was my pleasure!
Still, there is an issue in the online document. The hyperlink of the equations 1 and 2 are wrong. They point to a different document (i,e., https://pythonrobotics.readthedocs.io/en/latest/modules/path_planning/path_planning.html#equation-eq1)

image

**Comment by AtsushiSakai** [comment]
Thanks, I filed it as #632

**Pull request merged**